tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -370,22 +387,21 @@ runTest(ID3D12Device *D3DDevice, bool VerboseLogging,
       Operation, OpDataType, OutDataType, Inputs[0].size());
 
   dxc::SpecificDllLoader DxilDllLoader;
-
-  // The name of the shader we want to use in ShaderOpArith.xml. Could also
-  // add logic to set this name in ShaderOpArithTable.xml so we can use
-  // different shaders for different tests.
-  LPCSTR ShaderName = "LongVectorOp";
-  // ShaderOpArith.xml defines the input/output resources and the shader
-  // source.
   CComPtr<IStream> TestXML;
   readHlslDataIntoNewStream(L"ShaderOpArith.xml", &TestXML, DxilDllLoader);
+  auto ShaderOpSet = std::make_shared<st::ShaderOpSet>();
+  st::ParseShaderOpSetFromStream(TestXML, ShaderOpSet.get());
+
+  if (LoadAndStoreOpTypes.count(Operation.Type) > 0)
+    configureLoadAndStoreShaderOp(Operation, OpDataType, Inputs[0].size(),
+                                  sizeof(T), ShaderOpSet);
 
   // RunShaderOpTest is a helper function that handles resource creation
   // and setup. It also handles the shader compilation and execution. It takes
   // a callback that is called when the shader is compiled, but before it is
   // executed.
-  std::shared_ptr<st::ShaderOpTestResult> TestResult = st::RunShaderOpTest(
-      D3DDevice, DxilDllLoader, TestXML, ShaderName,
+  auto TestResult = st::RunShaderOpTestAfterParse(
```

Comment:
I think it was better to be explicit about the typename here.  I can't tell from just reading the code here what the return type of `RunShaderOpTestAfterParse` is.